runtime/scripts/jme-builtins.js

```diff
@@ -941,22 +941,178 @@ jme.substituteTreeOps.let = function(tree,scope,allowUnbound) {
         tree.args[i] = jme.substituteTree(tree.args[i],scope,allowUnbound);
     }
 }
-newBuiltin('sort',[TList],TList, null, {
-    evaluate: function(args,scope)
+var sortCompare = function(a, b) {
+    // We can't compare complex numbers
+    if (a.complex || b.complex) {
+        throw new Numbas.Error('math.order complex numbers');
+    }
+    return (a.value - b.value);
+};
+newBuiltin('sort', [TList], TList, null, {
+    evaluate: function(args, scope)
     {
-        var list = args[0];
-        var newlist = new TList(list.vars);
-        newlist.value = list.value.slice().sort(function(a,b){
-            if(math.gt(a.value,b.value))
-                return 1;
-            else if(math.lt(a.value,b.value))
-                return -1;
-            else
-                return 0;
+        var listCopy = args[0].value.slice();
+        listCopy.sort(sortCompare); // In-place
+        return new TList(listCopy);
+    }
+});
+var sortByHelper = function(list, accessor, extra) {
```

Comment:
`accessor` is not quite the right name for the second argument. It might be a function which transforms the element instead of just accessing one of its properties. `keyFn` might be better.